flood_forecast/custom/custom_opt.py

```diff
@@ -14,27 +16,91 @@ def warmup_cosine(x, warmup=0.002):
         return x/warmup
     return 0.5 * (1.0 + torch.cos(math.pi * x))
 
+
 def warmup_constant(x, warmup=0.002):
     """ Linearly increases learning rate over `warmup`*`t_total` (as provided to BertAdam) training steps.
         Learning rate is 1. afterwards. """
     if x < warmup:
         return x/warmup
     return 1.0
 
+
 def warmup_linear(x, warmup=0.002):
     """ Specifies a triangular learning rate schedule where peak is reached at `warmup`*`t_total`-th (as provided to BertAdam) training step.
         After `t_total`-th training step, learning rate is zero. """
     if x < warmup:
         return x/warmup
     return max((x-1.)/(warmup-1.), 0)
 
+
 SCHEDULES = {
     'warmup_cosine':   warmup_cosine,
     'warmup_constant': warmup_constant,
     'warmup_linear':   warmup_linear,
 }
 
 
+class RMSELoss(torch.nn.Module):
+    '''
+    Returns RMSE using:
+    target -> True y
+    output -> Predtion by model
+    source: https://discuss.pytorch.org/t/rmse-loss-function/16540/3
+    '''
+    def __init__(self):
+        super().__init__()
+        self.mse = torch.nn.MSELoss()
+
+    def forward(self, target: torch.Tensor, output: torch.Tensor):
+        return torch.sqrt(self.mse(target, output))
+
+
+class MAPELoss(torch.nn.Module):
+    '''
+    Returns MAPE using:
+    target -> True y
+    output -> Predtion by model
+    '''
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, target: torch.Tensor, output: torch.Tensor):
+        return torch.mean(torch.abs((target - output) / target))
+    
+# Add custom loss function
+class GaussianLoss(torch.nn.Module):
+    def __init__(self,mu,sigma):
+        """Compute the negative log likelihood of Gaussian Distribution
+        From https://arxiv.org/abs/1907.00235
+        """
+        super(GaussianLoss, self).__init__()
```

Comment:
I don't think you need `super(GaussianLoss, self).__init__()` in python 3, although it's still valid. Probably we should just stick with one way of doing it rather than have 2 different ways of calling the super class in the same file...

Reply:
Yeah this was just a copy from the original code of the authors. But I'll fix it